### ISSUE TITLE
Custom exception handling has inconsistent behaviour between dev/test and prod environment

### ISSUE THREAD

**Issue by odoucet** [api-platform/core#4275]
**API Platform version(s) affected**: 2.6.4 and lower (at least)

**Description**  
If you register a custom Exception in config with `api_platform.exception_to_status` and a status code >= 500 and < 600, trait `ApiPlatform\Core\Problem\Serializer\ErrorNormalizerTrait` has a very special handling that may prevent code from working in production environment (and it works on dev/test mode).

Problem is in getErrorMessage : 
```
// ApiPlatform\Core\Problem\Serializer/ErrorNormalizerTrait
private function getErrorMessage($object, array $context, bool $debug = false): string
    {
        $message = $object->getMessage();

        if ($debug) {
            return $message;
        }

        if ($object instanceof FlattenException || $object instanceof LegacyFlattenException) {
            $statusCode = $context['statusCode'] ?? $object->getStatusCode();
            if ($statusCode >= 500 && $statusCode < 600) {
                $message = Response::$statusTexts[$statusCode];
            }
        }

        return $message;
    }
```
There is no check `Response::$statusTexts[$statusCode]` exists, and in this case, another error (TypeError) is raised and prevent code from working. 
I also do not understand why there is a special shortcut in debug mode.

**How to reproduce**  
in api_platform.yml : 
```
api_platform:
    exception_to_status:
        App\Exception\CustomException: 509
```
Note that I used a 509 HTTP code, that is not an official code.

File src\Exception\CustomException.php : 
```
<?php
namespace App\Exception;
class CustomException extends \Exception
{
}
```

**Possible Solution**  
There is two possible fixes and I need some feedback before creating a pull request : 
1) First, explain to me why there is a shortcut in debug mode. Why would we want a different message between envs ? This is really confusing.
2) then, we should check if `Response::$statusTexts[$statusCode]` exists and return a generic error message if not.

**Comment by alanpoulain** [comment]
Hello Olivier.
For your question, the answer is really simple: it's because we don't want to show the message in production to the users, because it can leak important information.
Yes, it could be a good idea to do this. Do you want to work on this PR?

**Comment by alanpoulain** [comment]
That's right. It's because 500+ errors are often unwanted errors whereas for instance 400+ errors (validation, ...) are thought to be shown to the end user.

**Comment by odoucet** [comment]
I've open both MR.
A test would be great on #4277 but I really don't know how to handle this in APIP test framework.
